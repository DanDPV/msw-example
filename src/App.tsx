import Character from 'components/Character';
import RandomQuote from 'components/RandomQuote';
import React from 'react';

function App() {
  return (
    <div>
      <h1>Breaking bad API</h1>
      <RandomQuote />
      <Character id={1} />
    </div>
  );
}

export default App;
