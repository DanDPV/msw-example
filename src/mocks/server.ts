/* eslint-disable import/extensions */
/* eslint-disable import/no-extraneous-dependencies */
import { setupServer } from 'msw/node';
import handlers from './handlers';
// This configures a request mocking server with the given request handlers.
const server = setupServer(...handlers);

export default server;
