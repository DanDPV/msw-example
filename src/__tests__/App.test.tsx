/* eslint-disable no-undef */
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import App from 'App';
import server from 'mocks/server';
import { rest } from 'msw';

describe('Test on App component', () => {
  test('should render app correctly', async () => {
    const { container, getByText } = render(<App />);

    await waitFor(() => {
      expect(
        getByText("You forgot your ice trays! YOU'RE GONNA NEED THE ICE TRAYS!"),
      ).toBeInTheDocument();
      expect(getByText('Walter White (Heisenberg)')).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  test('should handle errors', async () => {
    server.use(
      rest.get(
        `${process.env.REACT_APP_API_URL}quote/random`,
        (req, res, ctx) => res(ctx.status(404)),
      ),

      rest.get(
        `${process.env.REACT_APP_API_URL}characters/:idCharacter`,
        (req, res, ctx) => res(ctx.status(404)),
      ),
    );

    const { getByText } = render(<App />);

    await waitFor(() => {
      expect(getByText('Could not load quote...')).toBeInTheDocument();
      expect(getByText('Could not load character...')).toBeInTheDocument();
    });
  });
});
