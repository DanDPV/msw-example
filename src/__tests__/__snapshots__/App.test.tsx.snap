// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on App component should render app correctly 1`] = `
<div>
  <div>
    <h1>
      Breaking bad API
    </h1>
    <div>
      <h2>
        <b>
          Jesse Pinkman
        </b>
      </h2>
      <p>
        You forgot your ice trays! YOU'RE GONNA NEED THE ICE TRAYS!
      </p>
    </div>
    <div>
      <h2>
        <b>
          Walter White (Heisenberg)
        </b>
      </h2>
      <img
        alt="character"
        src="https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg"
        style="width: 15%;"
      />
      <p>
        High School Chemistry Teacher
        Meth King Pin
      </p>
    </div>
  </div>
</div>
`;
